lib/puppetserver/utils/file_utils.rb

```diff
@@ -0,0 +1,64 @@
+require 'fileutils'
+
+module Puppetserver
+  module Utils
+    module FileUtilities
+
+      def self.ensure_dir(setting)
+        if !File.exist?(setting)
+          FileUtils.mkdir_p(setting, mode: 0750)
+          user, group = find_user_and_group
+          FileUtils.chown(user, group, setting)
+        end
+      end
+
+      def self.ensure_file(location, content, mode)
+        if !File.exist?(location)
+          Puppetserver::Utils::FileUtilities.write_file(location, content, mode)
+        end
+      end
+
+      def self.validate_file_paths(one_or_more_paths)
+        return [] unless one_or_more_paths.any?
+        errors = []
+        Array(one_or_more_paths).each do |path|
```

Comment:
If we're going to require that the input to the method be an array we can change the name of the variable and remove the coercion with Array here.

If we're going to allow a single string path (previous functionality), then we should either remove the call to any? above or wrap it in a coercion as well.

I'm not sure I think the way this changed from the previous implementation - from allowing a single string path or (a potentially empty) array of string paths, to requiring an array of either string paths or falsey values - is an improvement.

I'm fine requiring the caller to provide an array of values, but I'm concerned about requiring it to be either all valid string paths, or all falsey values. I'd prefer to require the caller to do its nil checks as a mix of valid paths and nils will cause issues below.

Or, potentially, allow a single nil value (remove the any? from the unless conditional)?
I guess, calling compact would be a way to allow for the caller to not care about a mix of nils and file paths. But I still think that it would be more supportable long term to require the caller to do its nil checks (like I would expect a nil value passed in here to cause an error).

Reply:
fixed! thanks